### PR TITLE
Initial upload of blob TTL template

### PR DESCRIPTION
**Pull request by yzheng-msft**

### CODE REVIEW

File: templates/Azure-Blob-Auto-Expiration.json [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                    },
+                    "method": "delete",
+                    "path": "/datasets/default/files/@{encodeURIComponent(encodeURIComponent(items('For_each')?['Id']))}"
+                  },
+                  "runAfter": {},
+                  "type": "ApiConnection"
+                }
+              },
+              "description": "Check LastModified timestamp and whether older than the expiration age variable",
+              "expression": "@less(items('For_each')?['LastModified'], addDays(utcnow(), variables('ExpirationAgeInDays')))",
```
Comment (derek1ee) [inline templates/Azure-Blob-Auto-Expiration.json]: Here you're essentially doing a string comparison on two values that represent DateTime, will they always be of the same format? I think it's better to convert them to ticks before doing such comparison.

E.g. `@less(ticks(items('For_each')?['LastModified']), ticks(addDays(utcnow(), variables('ExpirationAgeInDays')))`

File: templates/Azure-Blob-Auto-Expiration.json [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        },
+        "List_blobs": {
+          "inputs": {
+            "host": {
+              "connection": {
+                "name": "@parameters('$connections')['azureblob']['connectionId']"
+              }
+            },
+            "method": "get",
+            "path": "/datasets/default/foldersV2/@{encodeURIComponent(encodeURIComponent('L3Rlc3Q='))}",
```
Comment (derek1ee) [inline templates/Azure-Blob-Auto-Expiration.json]: I think the value should be `/datasets/default/foldersV2/`, without the url encoding part?

File: templates/Azure-Blob-Auto-Expiration.json [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            "queries": {
+              "useFlatListing": true
+            }
+          },
+          "metadata": {
```
Comment (derek1ee) [inline templates/Azure-Blob-Auto-Expiration.json]: Needed?

File: templates/Azure-Blob-Auto-Expiration.json [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+          "id": "/subscriptions/{0}/providers/Microsoft.Web/locations/{1}/managedApis/azureblob"
+        }
+      }
+    },
+    "apiSummaries": [
+      {
+        "type": "ApiConnection",
+        "displayName": "Azure Blob",
+        "iconUri": "https://cpgeneralstore.blob.core.windows.net/icons/azureblob.png",
+        "brandColor": "#0072c6"
```
Comment (derek1ee) [inline templates/Azure-Blob-Auto-Expiration.json]: #804998

File: templates/Azure-Blob-Auto-Expiration.json [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+      "azureblob": {
+        "connection": {
+          "id": ""
+        },
+        "api": {
+          "id": "/subscriptions/{0}/providers/Microsoft.Web/locations/{1}/managedApis/azureblob"
+        }
+      }
+    },
+    "apiSummaries": [
```
Comment (derek1ee) [inline templates/Azure-Blob-Auto-Expiration.json]: Should include a "Recurrence" object:

```
{
                "type": "recurrence",
                "displayName": "${Resources.RECURRENCE_TRIGGER_DISPLAYNAME}",
                "iconUri": "${Images.RecurrenceIcon}",
                "brandColor": "#1F85FF"
            }
```

File: templates/Azure-Blob-Auto-Expiration.json (hunk 1 above)
Comment (yzheng-msft) [reply templates/Azure-Blob-Auto-Expiration.json]: fixed

File: templates/Azure-Blob-Auto-Expiration.json (hunk 2 above)
Comment (yzheng-msft) [reply templates/Azure-Blob-Auto-Expiration.json]: fixed

File: templates/Azure-Blob-Auto-Expiration.json (hunk 3 above)
Comment (yzheng-msft) [reply templates/Azure-Blob-Auto-Expiration.json]: fixed

File: templates/Azure-Blob-Auto-Expiration.json (hunk 5 above)
Comment (yzheng-msft) [reply templates/Azure-Blob-Auto-Expiration.json]: fixed

File: templates/Azure-Blob-Auto-Expiration.json (hunk 4 above)
Comment (yzheng-msft) [reply templates/Azure-Blob-Auto-Expiration.json]: fixed

**Pull request merged**